data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -82,15 +113,36 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
                 throw new PatternSyntaxException("key_value_delimiter_regex is not a valid regex string", keyValueProcessorConfig.getKeyValueDelimiterRegex(), -1);
             }
 
+            if (keyValueProcessorConfig.getRecursive()) {
+                if (Pattern.compile(keyValueProcessorConfig.getKeyValueDelimiterRegex()).matcher(delimiterBracketCheck).matches()) {
+                    throw new IllegalArgumentException("The set key value delimiter regex cannot contain brackets while you are trying to recurse.");
+                }
+                if (keyValueProcessorConfig.getKeyValueDelimiterRegex().length() != 1) {
+                    throw new IllegalArgumentException("The set key value delimiter regex is limited to one character only.");
+                }
+            }
+
             keyValueDelimiterPattern = Pattern.compile(keyValueProcessorConfig.getKeyValueDelimiterRegex());
         } else {
             String regex;
-            if(keyValueProcessorConfig.getValueSplitCharacters().isEmpty()) {
+            if (keyValueProcessorConfig.getValueSplitCharacters().isEmpty()) {
                 regex = KeyValueProcessorConfig.DEFAULT_VALUE_SPLIT_CHARACTERS;
             } else {
+                if (keyValueProcessorConfig.getRecursive()) {
+                    if (keyValueProcessorConfig.getValueSplitCharacters().length() != 1) {
+                        throw new IllegalArgumentException("The set value split characters is limited to one character only.");
```

Comment:
Similar as above 

`The set value split characters is limited to one character only when recursive is true`